ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java

```diff
@@ -512,6 +513,9 @@ private String toString(Map<String, String> parameters) {
         if (dpPartSpecs != null) {
           // load the list of DP partitions and return the list of partition specs
           list.addAll(dpPartSpecs);
+          // Reload partition metadata because another BasicStatsTask instance may have updated the stats.
+          List<String> partNames = list.stream().map(Partition::getName).collect(Collectors.toList());
```

Comment:
Initially I didn't want to refactor this method just provide a simple fix to the correctness issue.
Thanks for your suggestions these are small changes and makes sense to me so I'm open to expand a bit the scope of this patch.

My last commit is also a refactor: since the stream is created over the `dbPartSpecs` the `list` object became unnecessary in this if branch. So I decided to remove it completely.
Let's see how ptest performs with this change.